### PR TITLE
Rebuild for numpy 2.0

### PR DESCRIPTION
**Pull request by h-vetinari**
Not sure what the bot stumbled over, but since this is one of the biggest feedstocks in the migration, let's fix it rather than wait.

Fixes https://github.com/conda-forge/pandas-feedstock/issues/204

### CODE REVIEW

**Comment by h-vetinari** [comment]
Travis hangs despite a restart, and is just generally a horrible provider to work with (the most basic UX is broken). The PPC builds run in emulation anyway, so no reason not to do aarch there as well (as the time-to-finish-CI is unaffected).

**Comment by lithomas1** [comment]
I worry about the switching off of Travis part. 

ppc64le emulation is really slow, and aarch64 emulation used to be really slow from what I recall.
(ppc64le takes like 5 hours, and if aarch takes the same amount of time, it would make releases take way to long for me)

On releases, I normally just merge without ppc64le since i don't care about it, but aarch64 actually has a decent chunk of users using it, so that's probably not a good idea.

Is there any other aarch64 provider that conda-forge has?

On the main pandas repo, we use CircleCI, but not sure if that's available here.

**Comment by jakirkham** [comment]
For clarity, Axel's changes still produce ARM & PPC packages. They just use cross-compilation instead of emulation

Should add cross-compilation is much faster than emulation. It is equivalent to native build time

The main trade off with cross-compilation is we lose the ability to run the tests

**Comment by h-vetinari** [comment]
> I worry about the switching off of Travis part.

The build hangs on Travis, so it's not like it's of much use.

> ppc64le emulation is really slow, and aarch64 emulation used to be really slow from what I recall.

They take +/- the exact same amount of time.

> if aarch takes the same amount of time, it would make releases take way too long for me

Releases happen rarely, so I don't really see the constraint (apart from the fact that ppc already takes that long).

> On releases, I normally just merge without ppc64le since i don't care about it

This is not how you should treat this or any other feedstock. But let's assume we don't care for now, then you could simply switch off the test suite on aarch/ppc, which is the part that actually takes a long time. Assuming you have upstream CI for aarch64, this shouldn't matter (kinda like the test suite already isn't being run for pypy).

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,6 +30,7 @@ requirements:
     - meson-python                           # [build_platform != target_platform]
     - ninja                                  # [build_platform != target_platform]
     - {{ compiler('c') }}
+    - {{ stdlib('c') }}
     - {{ compiler('cxx') }}
   host:
     - python
```
Comment (jorisvandenbossche) [inline recipe/meta.yaml:36]: For a few lines below, should `{{ pin_compatible('numpy') }}` be change to just `numpy` ?

Comment (h-vetinari) [reply recipe/meta.yaml:36]: Thanks for catching that, I overlooked it while doing it manually. It should be removed.

**Comment by lithomas1** [comment]
> 
> > ppc64le emulation is really slow, and aarch64 emulation used to be really slow from what I recall.
> 
> They take +/- the exact same amount of time.
> 
> > if aarch takes the same amount of time, it would make releases take way too long for me
> 
> Releases happen rarely, so I don't really see the constraint (apart from the fact that ppc already takes that long).

I release around once a month, so yes this is important to me.
Ideally, I'd like to start/finish a release within the same day, which doesn't happen if I have to wait the full time for ppc64le.

> 
> > On releases, I normally just merge without ppc64le since i don't care about it
> 
> This is not how you should treat this or any other feedstock. But let's assume we don't care for now, then you could simply switch off the test suite on aarch/ppc, which is the part that actually takes a long time. Assuming you have upstream CI for aarch64, this shouldn't matter (kinda like the test suite already isn't being run for pypy).

That's a good idea. Do you mind doing that here?

**Comment by h-vetinari** [comment]
OK, putting this in to continue advancing in the migration; feel free to modify further to your taste of course.

**Comment by lithomas1** [comment]
Thanks for the quick PR!

**Pull request merged**